uk.co.compucorp.civicrm.hrcore/gulpfile.js

```diff
@@ -46,6 +47,62 @@ var Promise = require('es6-promise').Promise;
     });
   });
 
+  /**
+   * Constructs URL for BackstopJS scenario based on
+   * site URL, scenario config URL and contact "roles" and IDs map
+   *
+   * @param  {String} siteUrl
+   * @param  {String} scenarioUrl
+   * @param  {Object} contactIdsByRoles
+   * @return {String}
+   */
+  function constructBackstopJSScenarioUrl (siteUrl, scenarioUrl, contactIdsByRoles) {
+    return siteUrl + '/' +
+      scenarioUrl.replace(/\{\{contactId:([^}]+)\}\}/g, function (fullMatch, contactRole) {
+        return contactIdsByRoles[contactRole];
+      });
+  }
+
+  /**
+   * Fetches civicrm contacts who's emails match "civihr_" pattern
+   * and returns a map of their "roles" connected to their IDs.
+   * Requires 'civihr_(staff|manager|admin)@...' to be presented in DB,
+   * otherwise will throw an error.
+   *
+   * @return {Promise} resolved with {Object}, ex. { 'staff': 204, ... etc }
+   */
+  function getRolesAndIDs () {
+    return new Promise(function (resolve, reject) {
+      exec('cv api contact.get sequential=1 contact_type="Individual" return="email,contact_id"', function (err, result) {
+        var idsByRoles, missingRoles;
+
+        if (err) {
+          return reject(new Error('Unable to fetch contact roles and IDs: ' + err));
+        }
+
+        idsByRoles = _(JSON.parse(result).values)
+          .filter(function (contact) {
+            return contact.email.match(/^civihr_/);
+          })
+          .map(function (contact) {
+            return [contact.email.split('@')[0].split('_')[1], contact.contact_id];
```

Comment:
put `contact.email.split('@')[0].split('_')[1]` in a descriptive variable, it increases readability 
```js
.map(function (concat) {
  var descriptiveVar = contact.email.split('@')[0].split('_')[1];

  return [descriptiveVar, contact.contact_id];
});
```

Reply:
Done.